feat(SDRS): add a new resource sdrs replication pair (#3121)
huaweicloud/terraform-provider-huaweicloud: docs/resources/sdrs_replication_pair.md

---
subcategory: "Storage Disaster Recovery Service (SDRS)"
---

# huaweicloud_sdrs_replication_pair

Manages an SDRS replication pair resource within HuaweiCloud.

## Example Usage

```hcl

> data "huaweicloud_availability_zones" "test" {}
> data "huaweicloud_sdrs_domain" "test" {}
> data "huaweicloud_vpc" "test" {
>   name = "vpc-default"
> }
> 
> resource "huaweicloud_sdrs_protectiongroup" "test" {
>   name                     = "test-name"
>   source_availability_zone = data.huaweicloud_availability_zones.test.names[0]
>   target_availability_zone = data.huaweicloud_availability_zones.test.names[1]
>   domain_id                = data.huaweicloud_sdrs_domain.test.id
>   source_vpc_id            = data.huaweicloud_vpc.test.id
>   description              = "test description"
> }

Reviewer: ```suggestion
variable "source_availability_zone" {}
variable "target_availability_zone" {}
variable "domain_id" {}
variable "vpc_id" {}

resource "huaweicloud_sdrs_protectiongroup" "test" {
  name                     = "test-name"
  source_availability_zone = var.source_availability_zone
  target_availability_zone = var.target_availability_zone
  domain_id                = var.domain_id
  source_vpc_id            = var.vpc_id
  description              = "test description"
}
```

It is better to define parameters as variables to highlight the way resource scripts are written.

resource "huaweicloud_evs_volume" "test" {
  name              = "test-evs-name"
  description       = "test volume for sdrs replication pair"

>   availability_zone = data.huaweicloud_availability_zones.test.names[0]

Reviewer: ditto

  volume_type       = "SSD"
  size              = 100
}

resource "huaweicloud_sdrs_replication_pair" "test" {
  name                 = "test-replication-pair"
  group_id             = huaweicloud_sdrs_protectiongroup.test.id
  volume_id            = huaweicloud_evs_volume.test.id
  description          = "test description"
  delete_target_volume = true
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

> * `name` - (Required, String) Specifies the name of a replication pair. The name can contain a maximum of 64 bytes.

Reviewer: ```suggestion
* `name` - (Required, String) Specifies the name of a replication pair. The name can contain a maximum of 64 characters.
```

>   The value can contain only letters (a to z and A to Z), digits (0 to 9), decimal points (.), underscores (_),

Reviewer: ```suggestion
  The value can contain only letters (a to z and A to Z), digits (0 to 9), periods (.), underscores (_),
```

or 

```suggestion
  The value can contain only letters (a to z and A to Z), digits (0 to 9), dots (.), underscores (_),
```

  and hyphens (-).

* `group_id` - (Required, String, ForceNew) Specifies the ID of a protection group.

  Changing this parameter will create a new resource.

* `volume_id` - (Required, String, ForceNew) Specifies the ID of the production site disk.
  When the provider is successfully invoked, the disaster recovery site disk will be automatically created.

  Changing this parameter will create a new resource.

* `delete_target_volume` - (Optional, Bool, ForceNew) Specifies whether to delete the disaster recovery site disk.
  The default value is **false**.

  Changing this parameter will create a new resource.

* `description` - (Optional, String, ForceNew) Specifies the description of a replication pair. The value can contain

>   a maximum of 64 bytes and cannot contain the left angle bracket (<) or right angle bracket (>).

Reviewer: ```suggestion
  a maximum of 64 characters and angle brackets (<) and (>) are not allowed.
```
